unit-tests/src/test/java/org/eclipse/collections/impl/stack/mutable/ArrayStackTest.java

```diff
@@ -45,6 +46,13 @@ protected <T> MutableStack<T> newStack(Iterable<T> elements)
         return Stacks.mutable.ofAll(elements);
     }
 
+    @Test
+    public void distinct()
+    {
+        MutableStack<Object> arrayStack = new ArrayStack<>(1, 1, 2, 2, 3, 3, 4, 4, 5, 5);
+        Verify.assertListsEqual(arrayStack.distinct().toList(), this.newStackWith(1, 2, 3, 4, 5).toList());
```

Comment:
```suggestion
        MutableStack<Object> stack = new ArrayStack<>(1, 1, 2, 2, 3, 3, 4, 4, 5, 5);
        MutableStack<Object> actual   = stack.distinct();
        MutableStack<Integer> expected  = this.newStackWith(1, 2, 3, 4, 5);
        assertEquals(expected, actual);
```

Reply:
We can assert that the stacks are equal rather than their lists.

Reply:
We can refactor this to be a bit more generic and pull it up to the superclass StackIterableTestCase.

```
    @Test
    public void distinct()
    {
        MutableStack<Object> stack = this.newStackFromTopToBottom(5, 5, 4, 4, 3, 3, 2, 2, 1, 1);
        MutableStack<Object> actual   = stack.distinct();
        MutableStack<Integer> expected  = this.newStackWith(1, 2, 3, 4, 5);
        assertEquals(expected, actual);
    }
```

Reply:
I've changed MutableStack<Object> to StackIterable<Integer> and pulled it upto StackIterableTestCase class